nvtabular/io/dataset.py

```diff
@@ -688,24 +698,40 @@ def to_parquet(
         # Convert `output_files` argument to a dict mapping
         if output_files:
 
-            # First, repartition ddf if necessary
+            # Use out_files_per_proc to calculate how
+            # many output files should be written within the
+            # same subgraph.  Note that we must a
+            files_per_task = out_files_per_proc or 1
             required_npartitions = ddf.npartitions
             if isinstance(output_files, int):
                 required_npartitions = output_files
-            elif isinstance(output_files, list):
+                files_per_task = min(files_per_task, output_files)
+            elif isinstance(output_files, int):
```

Comment:
Is this right? Looks like this  should still be checking for a list (like we're checking isinstance int on line 706)
```suggestion
            elif isinstance(output_files, list):
```

Reply:
Oops - good catch :)